pkg/formulaexec/formula_exec.go

```diff
@@ -195,25 +203,72 @@ func getBaseConfig(ctx context.Context, wsPath, runPath, binPath string) (runCon
 			"--rootless",
 			"-b", runPath)
 	}
-	err = cmd.Run()
+	err := cmd.Run()
 	tracing.EndWithStatus(cmdSpan, err)
 	if err != nil {
-		return rc, wfapi.ErrorExecutorFailed("failed to generate runc config", err)
+		return result, wfapi.ErrorExecutorFailed("failed to generate runc config", err)
 	}
+
 	configFileBytes, err := ioutil.ReadFile(configFile)
 	if err != nil {
-		return rc, wfapi.ErrorIo("failed to read runc config", configFile, err)
+		return result, wfapi.ErrorIo("failed to read runc config", configFile, err)
 	}
-	err = json.Unmarshal(configFileBytes, &rc.spec)
+
+	err = json.Unmarshal(configFileBytes, &result)
 	if err != nil {
-		return rc, wfapi.ErrorExecutorFailed("runc",
+		return result, wfapi.ErrorExecutorFailed("runc",
 			wfapi.ErrorSerialization("failed to parse runc config", err))
 	}
+	return result, nil
+}
 
-	// set up root -- this is not actually used since it is replaced with an overlayfs
-	rootPath := filepath.Join(runPath, "root")
-	err = os.MkdirAll(rootPath, 0755)
+// copySpec returns a copy of the spec
+// Internally, copySpec serializes and deserializes the data
+// Presumably this can round trip, but an error is returned just in case.
+//
+// Errors:
+//
+//    - warpforge-error-internal -- when copying the spec fails
+func copySpec(spec specs.Spec) (specs.Spec, error) {
+	var result specs.Spec
+	raw, err := json.Marshal(spec)
+	if err != nil {
+		return result, serum.Error(wfapi.ECodeInternal,
+			serum.WithCause(wfapi.ErrorSerialization("failed to serialize OCI spec during copy", err)),
+		)
+	}
+	err = json.Unmarshal(raw, &result)
+	if err != nil {
+		return result, serum.Error(wfapi.ECodeInternal,
+			serum.WithCause(wfapi.ErrorSerialization("failed to parse OCI spec during copy", err)),
+		)
+	}
+	return result, nil
+}
+
+// Creates a base configuration for runc, which is later modified before running``
+//
+// Errors:
+//
+//    - warpforge-error-io -- when file reads, writes, and dir creation fails
+//    - warpforge-error-internal -- copying the spec fails
+func getBaseConfig(ctx context.Context, wsPath, runPath, binPath string, warehousePathOverride lookupFunc, baseSpec specs.Spec) (runConfig, error) {
+	rc := runConfig{
+		runPath:     runPath,
+		wsPath:      wsPath,
+		binPath:     binPath,
+		interactive: false,
+	}
+	spec, err := copySpec(baseSpec)
 	if err != nil {
+		return rc, err
+	}
+	rc.spec = spec
+
+	// set up root -- this is not actually used since it is replaced with an overlayfs
+	rootPath := filepath.Join(runPath, "rooti") // FIXME: "root" is created as a regular file in the .warpforge dir? I think I screwed something up.
```

Comment:
rooti?  And is this fixme comment still living?

Reply:
hmmm well we'll find out if it explodes when I remove the `i`.